### DISCUSSION TITLE
XPath relative not working (android only tested)

### DISCUSSION THREAD

**Discussion by v-mwalk** [appium/appium#16513] [Q&A] [answered]
This has taken some digging, and I'm not 100% sure it is an Appium issue (issue is possibly with whatever is processing the XPath queries).  I have only seen this against Android as that is what I am currently writing automation for - I've not tried it against an iOS app or web.

I have XML (pseudo for readability):-
```
.
.
.
<parent>
  <child1/>
  <child2/>
  <child3/>
</parent>
.
.
.
```
const parent = await $('//parent');
const parentClass = await parent.getAttribute('class');
const allChildren = await parent.$$('./*');
const childrenLength = allChildren.length;
const child0Class = allChildren[0].getAttribute('class');

i would expect allChildrenLength to be 3 - it is 1.
i would expect child0Class to 'child1' but it is 'parent'

Either I'm doing something fundamentally wrong (95% sure I'm not as I've been using XPath's for far too many years to count :-) ) or there is a bug in the XPath interpreter...

Could anyone confirm (IE. Repro this) if this is a bug or if I'm being daft and have missed something.

**Comment by KazuCocoa** [comment, marked as answer]
When enabled `limitXPathContextScope` via settings api https://github.com/appium/appium-uiautomator2-driver#settings-api ?
It is only available with Appium 2 and the latest UIA2 driver.

**Reply by v-mwalk** [reply]
That is exactly it, thanks @KazuCocoa 
Have not looked at Appium2 yet - too under the pump at the moment - but will try to have a Beta play and look forward to production release :-)

**Reply by v-mwalk** [reply]
Tested using 2.0.0.-beta25 and it works a treat.  This also fixes issues such as using a relative path to locate sibling elements (IE. iterating through a RecyclerView set of children.

**Comment by PavithraRavichandran** [comment]
I face same issue with iOS app. Is there any work around for it? Please suggest
